spinalcordtoolbox/scripts/sct_register_to_template.py

```diff
@@ -332,18 +332,13 @@ def main(argv: Sequence[str]):
         smooth=arguments.centerline_smooth)
     # registration parameters
     if arguments.param is not None:
-        # reset parameters but keep step=0 (might be overwritten if user specified step=0)
+        # reset parameters but keep step=0 (can be overwritten if user specified step=0)
         paramregmulti = ParamregMultiStep([step0])
-        if ref == 'subject':
-            paramregmulti.steps['0'].dof = 'Tx_Ty_Tz_Rx_Ry_Rz_Sz'
         # add user parameters
         for paramStep in arguments.param:
             paramregmulti.addStep(paramStep)
     else:
         paramregmulti = ParamregMultiStep([step0, step1, step2])
-        # if ref=subject, initialize registration using different affine parameters
-        if ref == 'subject':
-            paramregmulti.steps['0'].dof = 'Tx_Ty_Tz_Rx_Ry_Rz_Sz'
```

Comment:
Thanks for this de-duplication! This code will be more straightforward to understand now. :hearts: